test/index.js

```diff
@@ -358,105 +281,64 @@ describe('Schwifty', () => {
-
-                expect(() => {
-
-                    server.register(plugin, () => done('Should not make it here.'));
-                }).to.throw('Schwifty\'s teardownOnStop option can only be specified once.');
-                // }).to.throw(/Schwifty\'s teardownOnStop option can only be specified once./);
+                    await srv.register({ options, plugin: Schwifty });
+                }
+            };
 
-                done();
-            });
+            await expect(server.register(plugin)).to.reject('Schwifty\'s teardownOnStop option can only be specified once.');
         });
 
-        it('throws when `migrateOnStart` is specified more than once.', (done) => {
-
-            getServer({ migrateOnStart: false }, (err, server) => {
-
-                expect(err).to.not.exist();
+        it('throws when `migrateOnStart` is specified more than once.', async () => {
 
-                const plugin = (srv, opts, next) => {
+            const server = await getServer({ migrateOnStart: false });
+            const plugin = {
+                name: 'my-plugin',
+                register: async (srv, opts) => {
 
-                    srv.register({ register: Schwifty, options: { migrateOnStart: false } }, next);
-                };
-
-                plugin.attributes = { name: 'my-plugin' };
-
-                expect(() => {
-
-                    server.register(plugin, () => done(new Error('Should not make it here.')));
-                }).to.throw('Schwifty\'s migrateOnStart option can only be specified once.');
+                    await srv.register({ plugin: Schwifty, options: { migrateOnStart: false } });
+                }
+            };
 
-                done();
-            });
+            await expect(server.register(plugin)).to.reject('Schwifty\'s migrateOnStart option can only be specified once.');
         });
 
-        it('throws when multiple knex instances passed to same server.', (done) => {
+        it('throws when multiple knex instances passed to same server.', async () => {
 
-            getServer({ knex: Knex(basicKnexConfig) }, (err, server) => {
+            const server = await getServer({ knex: Knex(basicKnexConfig) });
 
-                expect(err).to.not.exist();
-
-                expect(() => {
-
-                    server.register({
-                        register: Schwifty,
-                        options: { knex: Knex(basicKnexConfig) }
-                    }, (ignoreErr) => {
-
-                        return done(new Error('Should not make it here.'));
-                    });
-                }).to.throw('A knex instance/config may be specified only once per server or plugin.');
-
-                done();
-            });
+            await expect(server.register({
+                plugin: Schwifty,
+                options: { knex: Knex(basicKnexConfig) }
+            })).to.reject('A knex instance/config may be specified only once per server or plugin.');
```

Comment:
oh man, this prompted a dive into code's source to sort out why this works; I had no idea! 🎺  glad for that. thanks for fixing this 🙏